### ISSUE TITLE
Blog: enabling `use_directory_urls: false` breaks pagination

### ISSUE THREAD

**Issue by vedranmiletic** [squidfunk/mkdocs-material#5899]
### Bug description

Trying to build the minimal example with two post and pagination set to one post per page fails with:

```
ERROR   -  Error building page 'blog/index.md': [Errno 20] Not a directory: '/tmp/mkdocs_mktockyk/blog/index.html/page'
```

This is a regression compared to version 9.2.0b3, where it works correctly.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Setting up a blog](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/)

### Reproduction

[9.2.4-use_directory_urls-false-breaks-pagination.zip](https://github.com/squidfunk/mkdocs-material/files/12445802/9.2.4-use_directory_urls-false-breaks-pagination.zip)

### Steps to reproduce

1. run `mkdocs build`
2. observe the `ERROR` in console

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in fbb7a5e89. Additionally, I realized that the blog post title _Latest posts_ was not correctly carried over to paginated views. This was fixed in the same commit.

**Comment by squidfunk** [comment]
Okay, one more fix was necessary in 568a5c8e1 in order to preserve titles that are set from Markdown pages.

**Comment by squidfunk** [comment]
Released as part of [9.2.5](https://github.com/squidfunk/mkdocs-material/releases/tag/9.2.5).
